ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -426,6 +432,49 @@ void HttpOperation::Cleanup()
   }
 }
 
+bool HttpOperation::IsRetryable()
+{
+  static constexpr auto kRetryableStatusCodes = std::array<decltype(response_code_), 4>{
+      429,  // Too Many Requests
+      502,  // Bad Gateway
+      503,  // Service Unavailable
+      504   // Gateway Timeout
+  };
+
+  const auto is_retryable = std::find(kRetryableStatusCodes.cbegin(), kRetryableStatusCodes.cend(),
+                                      response_code_) != kRetryableStatusCodes.cend();
+
+  return is_retryable && (last_curl_result_ == CURLE_OK) &&
+         (retry_attempts_ < retry_policy_.max_attempts);
+}
+
+std::chrono::system_clock::time_point HttpOperation::NextRetryTime()
```

Comment:
The exponential backoff logic is modeled after the gRPC retry policy so that both OTLP gRPC and HTTP behave more or less consistently.